example/LockFreeQueue.chpl

```diff
@@ -0,0 +1,98 @@
+/**
+ * Created by Garvit Dewan - 
+ * https://github.com/dgarvit/epoch-based-manager/blob/master/src/LockFreeQueue.chpl
+ *   
+ * Lock-Free Queue that uses ABA feature of Distributed Data Structures
+ * Based on Michael Scott Queue
+ */
+module LockFreeQueue {
+
+  use LocalAtomics;
+
+  class node {
+    type eltType;
+    var val : eltType;
+    var next : LocalAtomicObject(unmanaged node(eltType));
+
+    proc init(val : ?eltType) {
+      this.eltType = eltType;
+      this.val = val;
+    }
+
+    proc init(type eltType) {
+      this.eltType = eltType;
+      val = nil;
```

Comment:
Do not initialize `val` with a value, leave it as is (it will be initialized to a default value automatically). If `eltType` is an `integral` (`int`,`uint`, etc.), a tuple or a record, this will result in a compiler error.

Reply:
Well, I think that would further require changes to `deinit()` in terms that types like `int` cannot be deleted;
To `peek()` and `dequeue()` in terms that if queue is empty then cannot return `nil` for `eltType` `int`;